pkg/segment/tracing/handler/tracehandler.go

```diff
@@ -515,44 +524,27 @@ func MakeTracesDependancyGraph() {
 }
 
 func ProcessDependencyRequest(ctx *fasthttp.RequestCtx, myid uint64) {
-	searchRequestBody := &structs.SearchRequestBody{}
-	searchRequestBody.QueryLanguage = "Splunk QL"
-	searchRequestBody.IndexName = "service-dependency"
-	searchRequestBody.SearchText = "*"
 
-	pipeSearchResponseOuter, err := processSearchRequest(searchRequestBody, myid)
-	if err != nil {
-		log.Errorf("ProcessSearchRequest: %v", err)
-		return
-	}
 	processedData := make(map[string]interface{})
-	if pipeSearchResponseOuter.Hits.Hits == nil || len(pipeSearchResponseOuter.Hits.Hits) == 0 {
+	depMatrix := MakeTracesDependancyGraph()
```

Comment:
This would make it so we never use our precomputed dependency graph. The reason for precomputing it is that computing it may take a while when there's lots of spans. Maybe we should change this so that if we have no precomputed graph then we compute it in this request (and write it to the index so we could use it for future requests), but if we do have precomputed graphs we just return the most recent precomputed one.